app/scripts/controllers/overview.js

```diff
@@ -498,6 +538,173 @@ angular.module('openshiftConsole')
 
         $scope.$on('$destroy', function(){
           DataService.unwatchAll(watches);
+          $(window).off('click.topologyItem');
+          $('.kube-topology g').popover('destroy');
         });
+
+        // Topology view specific code
+        var updateTimeout = null;
+
+        function updateTopology() {
+          updateTimeout = null;
+
+          var topologyRelations = [];
+          var topologyItems = { };
+
+          // Because metadata.uid is not unique among resources
+          function makeId(resource) {
```

Comment:
can `makeId` be moved up out of `updateTopology`? It prob only needs to be declared once & would shrink `updateTopology` a bit.